public/js/core.js

```diff
@@ -0,0 +1,193 @@
+angular.
+module('taskApp', []).
+controller('mainController', ['$scope', '$http', '$filter', 'getTaskFromWebSocket', function($scope, $http, $filter, getTaskFromWebSocket) {
+
+    $scope.formData = {};
```

Comment:
Do not use scope for this. 

Refactor all the code after reading https://toddmotto.com/digging-into-angulars-controller-as-syntax/